app/src/test/java/org/kiwix/kiwixmobile/zim_manager/ZimManageViewModelTest.kt

```diff
@@ -517,4 +517,18 @@ class ZimManageViewModelTest {
         .assertValues(StartMultiSelection(viewModel.fileSelectActions))
     }
   }
+
+  @Test
+  fun `finish actionMode on tab change to online section`() {
+    viewModel.libraryTabIsVisible.test()
+      .also { viewModel.currentPage.offer(1) }
+      .assertValue(1)
```

Comment:
@macgills here the filter condition wil pass and hence 1 value will be there. Right now I have hardcoded it. I'm not sure what condition to check for here.

Reply:
The test name is inaccurate, this knows nothing about an action mode. `libraryTabIsVisible emits when currentPage is 1` is probably most descriptive.

Aside from that this is all good, the logic of the filter is tested. We could take it a bit further because it feels a little odd that `libraryTabIsVisible` tells us `1` and also that we have no true concept in code that the `LibraryTab` is page 1, it is largely incidental.

To solve the `libraryTabIsVisible` problem is quite straighforward, we could map it

```
currentPage.map{ it == 1 }.filter { it }
```

now we would receive `true` here and continue to receive nothing in our other test.

The `1` problem requires us to model this data better. I tend to favour modelling tabs as Enums because enums give us access to a lot of fun little functions eg given an enum

```
enum class Suits(val color:String){
HEARTS("red"),DIAMONDS("red"),SPADES("black"),CLUBS("black")
}
```

if I were to call `Suits.values()` it would return a list of all those defined enums. Enums are also implicitly ordered eg `HEARTS.ordinal()==0`...`CLUBS.ordinal()==3`

Inside SectionsPagerAdapter we could migrate PagerData into being an enum, probably rename it to something more tabby too.

This is of course all quite a bit of overkill probably. If you want to learn more on enums etc, go for it, otherwise after a little rename of these test methods I am happy

Reply:
@macgills I have updated the pr with proper method names. Also, for now, I used a map and then a filter so that instead of `1` we could put `true` in the test to make it look more suitable. 

Thanks a lot for the above explanation. I'll look into what you told about modeling the data better using enums too later.

Reply:
@yashk2000 do you want to explore an enum solution now on this PR or should I just merge it?

Reply:
@macgills if it's fine, you can merge this.